### DISCUSSION TITLE
Graphql query returns null value for EXISTED field

### DISCUSSION THREAD

**Discussion by agsmartwolf** [swellstores/community#166] [Help] [answered]
Hi there,

I have a graphql query:
```
query getProduct($slug: String, $currency: String, $locale: String) {
  productBySlug(slug: $slug, _currency: $currency, _locale: $locale) {
    slug
    id
    name
    options {
      id
      attributeId
      name
      values {
        id
        name
        price
      }
    }
    stockLevel
  }
}
```

The field `attributeId` is returned as `null` value.
It started probably a week ago.
However, I can see this option 'attribute id' on the Swell dashboard in 'attributes' section.
<img width="950" alt="Screenshot 2023-03-06 at 20 48 35" src="https://user-images.githubusercontent.com/109105178/223176298-a821e7e3-6055-4632-97ba-320e0121e0f1.png">
<img width="1716" alt="image" src="https://user-images.githubusercontent.com/109105178/223176784-49cffc27-9d68-4e19-b0b0-a6b98c199f4a.png">

Could you help me to understand the problem?

**Comment by rodrigoddalmeida** [comment, marked as answer]
Hi @agsmartwolf, we're investigating this, as it looks like a potential bug in our core Backend API. We'll post updates here as it progresses.

**Reply by agsmartwolf** [reply]
Hi @rodrigoddalmeida, do you have any updates? It's a blocking bug to releasing my store to production right now.
If there are any temporary workarounds - I'd like to know about them.

**Reply by rodrigoddalmeida** [reply]
As a quick fix, you can toggle off `Automatically generate variants` (within Product page > Options > Edit) and save - toggle it back on to regenerate options and variants.

That should add missing `attribute_id` fields.

**Reply by agsmartwolf** [reply]
Are current variants will be removed when I toggle off it?
I have 154 variants for some of the products, and each one has its own uploaded image)))))
So It's pretty painful to remove them.

Attribute ids were already used earlier. This field was presented before in response but then disappeared.
Did you have some bad DB migration or some code that has overwritten this field for existing products?

I can try your suggested option for some of the products with a few variants.

**Reply by agsmartwolf** [reply]
I found a solution without removing current variants!!!

I've added a new attribute into option -> saved option -> saved product -> removed just created attribute -> saved option -> saved product.

**Reply by ericingram** [reply]
We generally don't do DB migrations that would affect store data, but glad you found a solution in any case!
